examples/msbr/openmc_msbr_model.py

```diff
@@ -0,0 +1,1045 @@
+            iec_b = (+gr_cyl_minx & -gr_cyl_maxx &
+            -gr_miny & +ul_t & +br & +gr_cyl_br)
+            iec_br = (+gr_cyl_br & +gr_cyl_maxx & -gr_cyl_miny) & +br & +lb_t
+
+            inter_elem_channel = [[iec_r, 'outer_r'], [iec_ru, 'outer_ru'],
+                                  [iec_u, 'outer_u'], [iec_ul, 'outer_ul'],
+                                  [iec_l, 'outer_l'], [iec_lb, 'outer_lb'],
+                                  [iec_b, 'outer_b'], [iec_br, 'outer_br']]
+        else:
+            # Graphite element rib regions.
+            rib_ul = -ul
+            rib_br = -br
+            rib_lb = -lb
+            rib_ru = -ru
+
+            # inter-element fuel channel region
+            inter_elem_channel = +ul & +br & +lb & +ru
+
+    elif elem_type == 'cr':
+        # Parameters for control rod element
+        r_d = 1.16
+        e_d = 2 * r_d / np.sqrt(3)
+        r_c = 0.18
+
+        # Base rib region
+        ul = openmc.model.hexagonal_prism(origin=(-l1, l2), edge_length=e_d,
+                                          orientation='x', corner_radius=r_c)
+        br = openmc.model.hexagonal_prism(origin=(l1, -l2), edge_length=e_d,
+                                          orientation='x',corner_radius=r_c)
+        lb = openmc.model.hexagonal_prism(origin=(-l2, -l1), edge_length=e_d,
+                                          orientation='y',corner_radius=r_c)
+        ru = openmc.model.hexagonal_prism(origin=(l2, l1), edge_length=e_d,
+                                          orientation='y',corner_radius=r_c)
+
+        if optimized:
+            # Split base rib region into component surfaces
+            (ul_u, ul_b, ul_ur, ul_br, ul_bl, ul_ul,
+             ul_cyl_bl_in, ul_cyl_bl_out,
+             ul_cyl_ul_in, ul_cyl_ul_out,
+             ul_cyl_br_in, ul_cyl_br_out,
+             ul_cyl_ur_in, ul_cyl_ur_out,
+             ul_cyl_l_in, ul_cyl_l_out,
+             ul_cyl_r_in, ul_cyl_r_out)= list(ul.get_surfaces().values())
+            (br_u, br_b, br_ur, br_br, br_bl, br_ul,
+             br_cyl_bl_in, br_cyl_bl_out,
+             br_cyl_ul_in, br_cyl_ul_out,
+             br_cyl_br_in, br_cyl_br_out,
+             br_cyl_ur_in, br_cyl_ur_out,
+             br_cyl_l_in, br_cyl_l_out,
+             br_cyl_r_in, br_cyl_r_out)= list(br.get_surfaces().values())
+            (lb_r, lb_l, lb_ur, lb_ul, lb_br, lb_bl,
+             lb_cyl_lb_in, lb_cyl_lb_out,
+             lb_cyl_lu_in, lb_cyl_lu_out,
+             lb_cyl_rb_in, lb_cyl_rb_out,
+             lb_cyl_ru_in, lb_cyl_ru_out,
+             lb_cyl_b_in, lb_cyl_b_out,
+             lb_cyl_u_in, lb_cyl_u_out)= list(lb.get_surfaces().values())
+            (ru_r, ru_l, ru_ur, ru_ul, ru_br, ru_bl,
+             ru_cyl_lb_in, ru_cyl_lb_out,
+             ru_cyl_lu_in, ru_cyl_lu_out,
+             ru_cyl_rb_in, ru_cyl_rb_out,
+             ru_cyl_ru_in, ru_cyl_ru_out,
+             ru_cyl_b_in, ru_cyl_b_out,
```

Comment:
This feels chaotic but it's not immediately obvious how to improve it. Is there a reason setting a single variable is not ideal?

Reply:
It is chaotic unfortunately. The purpose of this is to enable creation of regions that are only composed of intersections (no unions or region complements). This speeds up calculations by around 50%, but as you noticed, it's much harder to read.